lib/active_merchant/billing/gateways/pay_junction_v2.rb

```diff
@@ -0,0 +1,224 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class PayJunctionV2Gateway < Gateway
+      self.display_name = "PayJunction"
+      self.homepage_url = "https://www.payjunction.com/"
+
+      self.test_url = "https://api.payjunctionlabs.com/transactions"
+      self.live_url = "https://api.payjunction.com/transactions"
+
+      self.supported_countries = ["US"]
+      self.default_currency = "USD"
+      self.money_format = :dollars
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover]
+
+      def initialize(options={})
+        requires!(options, :api_login, :api_password, :api_key)
+        super
+      end
+
+      def purchase(amount, payment_method, options={})
+        post = {}
+        add_invoice(post, amount, options)
+        add_payment_method(post, payment_method)
+
+        commit("purchase", post)
+      end
+
+      def authorize(amount, payment_method, options={})
+        post = {}
+        post[:status] = "HOLD"
+        add_invoice(post, amount, options)
+        add_payment_method(post, payment_method)
+
+        commit("authorize", post)
+      end
+
+      def capture(amount, authorization, options={})
+        post = {}
+        post[:status] = "CAPTURE"
+        post[:transactionId] = authorization
+        add_invoice(post, amount, options) if amount
+
+        commit("capture", post)
+      end
+
+      def void(authorization, options={})
+        post = {}
+        post[:status] = "VOID"
+        post[:transactionId] = authorization
+
+        commit("void", post)
+      end
+
+      def refund(money, authorization, options={})
```

Comment:
@davidsantoso think I made a mistake in understanding PayJunction's `REFUND` transaction. thought it was allowing only AM `credit` style operation. re-reading the docs, looks like it supports both AM `credit` and AM `refund`

due to the misunderstanding I tried to accomplish a read-modify-write to implement AM `refund`. as in, read previously charged amount, reduce the amount to be refunded, and write back the new balance to be charged. but that seems a wrong approach now. thanks for spotting, let me rework this.